cmake/Dependencies.cmake

```diff
@@ -180,85 +179,63 @@ if (NOT INTERN_BUILD_MOBILE)
   endif()
 endif()
 
```

Comment:
Well as for the CMake changes, basically this first block below until my second comment in this file, factors out all the checks common to NNPACK and family into a unified block as opposed to repeating them over.  If any of these checks fail, NNPACK and family are disabled.  These conditions check whether the requested architecture and operating system are supported.